Comparison fails for unit tests with timestamps

This looks like #508, probably appearing again as the current logic checks for a string match of the `field.ident` and that changed in #851. 

```
________________________________________________ test_set_finding_state_flattened_async _________________________________________________

    @pytest.mark.asyncio
    async def test_set_finding_state_flattened_async():
        client = SecurityCenterAsyncClient(
            credentials=ga_credentials.AnonymousCredentials(),
        )
    
        # Mock the actual call within the gRPC stub, and fake the request.
        with mock.patch.object(
            type(client.transport.set_finding_state), "__call__"
        ) as call:
            # Designate an appropriate return value for the call.
            call.return_value = finding.Finding()
    
            call.return_value = grpc_helpers_async.FakeUnaryUnaryCall(finding.Finding())
            # Call the method with a truthy value for each flattened field,
            # using the keyword arguments to the method.
            response = await client.set_finding_state(
                name="name_value",
                state=finding.Finding.State.ACTIVE,
                start_time=timestamp_pb2.Timestamp(seconds=751),
            )
    
            # Establish that the underlying call was made with the expected
            # request object values.
            assert len(call.mock_calls)
            _, args, _ = call.mock_calls[0]
            assert args[0].name == "name_value"
            assert args[0].state == finding.Finding.State.ACTIVE
>           assert args[0].start_time == timestamp_pb2.Timestamp(seconds=751)
E           assert DatetimeWithNanoseconds(1970, 1, 1, 0, 12, 31, tzinfo=datetime.timezone.utc) == seconds: 751\n
E            +  where DatetimeWithNanoseconds(1970, 1, 1, 0, 12, 31, tzinfo=datetime.timezone.utc) = name: "name_value"\nstate: ACTIVE\nstart_time {\n  seconds: 751\n}\n.start_time
E            +  and   seconds: 751\n = <class 'google.protobuf.timestamp_pb2.Timestamp'>(seconds=751)
E            +    where <class 'google.protobuf.timestamp_pb2.Timestamp'> = timestamp_pb2.Timestamp

tests/unit/gapic/securitycenter_v1beta1/test_security_center.py:3670: AssertionError
```

--- a/gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
+++ b/gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
@@ -444,9 +444,9 @@ def test_{{ method.name|snake_case }}_flattened():
         assert len(call.mock_calls) == 1
         _, args, _ = call.mock_calls[0]
         {% for key, field in method.flattened_fields.items() %}{% if not field.oneof or field.proto3_optional %}
-        {% if field.ident|string() == 'timestamp.Timestamp' %}
+        {% if field.ident|string() == 'timestamp_pb2.Timestamp' %}
         assert TimestampRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
-        {% elif field.ident|string() == 'duration.Duration' %}
+        {% elif field.ident|string() == 'duration_pb2.Duration' %}
         assert DurationRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
         {% else %}
         assert args[0].{{ key }} == {{ field.mock_value }}

--- a/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
+++ b/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
@@ -760,9 +760,9 @@ def test_{{ method.name|snake_case }}_flattened():
         assert len(call.mock_calls) == 1
         _, args, _ = call.mock_calls[0]
         {% for key, field in method.flattened_fields.items() %}{% if not field.oneof or field.proto3_optional %}
-        {% if field.ident|string() == 'timestamp.Timestamp' %}
+        {% if field.ident|string() == 'timestamp_pb2.Timestamp' %}
         assert TimestampRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
-        {% elif field.ident|string() == 'duration.Duration' %}
+        {% elif field.ident|string() == 'duration_pb2.Duration' %}
         assert DurationRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
         {% else %}
         assert args[0].{{ key }} == {{ field.mock_value }}
@@ -845,9 +845,9 @@ async def test_{{ method.name|snake_case }}_flattened_async():
         assert len(call.mock_calls)
         _, args, _ = call.mock_calls[0]
         {% for key, field in method.flattened_fields.items() %}{% if not field.oneof or field.proto3_optional %}
-        {% if field.ident|string() == 'timestamp.Timestamp' %}
+        {% if field.ident|string() == 'timestamp_pb2.Timestamp' %}
         assert TimestampRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
-        {% elif field.ident|string() == 'duration.Duration' %}
+        {% elif field.ident|string() == 'duration_pb2.Duration' %}
         assert DurationRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
         {% else %}
         assert args[0].{{ key }} == {{ field.mock_value }}
@@ -1194,9 +1194,9 @@ def test_{{ method.name|snake_case }}_rest_flattened():
         body = http_params.get('data')
         {% for key, field in method.flattened_fields.items() %}
         {% if not field.oneof or field.proto3_optional %}
-        {% if field.ident|string() == 'timestamp.Timestamp' %}
+        {% if field.ident|string() == 'timestamp_pb2.Timestamp' %}
         assert TimestampRule().to_proto(http_call[0].{{ key }}) == {{ field.mock_value }}
-        {% elif field.ident|string() == 'duration.Duration' %}
+        {% elif field.ident|string() == 'duration_pb2.Duration' %}
         assert DurationRule().to_proto(http_call[0].{{ key }}) == {{ field.mock_value }}
         {% else %}
         assert {% if field.field_pb is msg_field_pb %}{{ field.ident }}.to_json({{ field.name }}, including_default_value_fields=False, use_integers_for_enums=False)

